common/src/main/kotlin/com/freeletics/gradle/plugin/FreeleticsGradlePluginPlugin.kt

```diff
@@ -31,35 +28,18 @@ abstract class FreeleticsGradlePluginPlugin : Plugin<Project> {
     }
 
     private fun Project.generateVersionTask() {
-        val directory = layout.buildDirectory.dir("generated/pluginVersion")
-
-        java {
-            sourceSets.findByName("main")!!.java.srcDir(directory)
-        }
-
-        val generateVersion = tasks.register("generatePluginVersion", GenerateVersion::class.java) { task ->
+        val generateVersion = tasks.register("generatePluginVersion", GenerateVersionClass::class.java) { task ->
+            val artifactId = property("POM_ARTIFACT_ID")!!.toString()
             val packageName = stringProperty("GROUP").map { group ->
-                val artifactId = property("POM_ARTIFACT_ID")!!.toString()
```

Comment:
The `map` gets serialized with the provider, so the `property("POM_ARTIFACT_ID")!!.toString()` call would access `Project` from within a task which fails the cache.